utils/revision_utils.go

```diff
@@ -49,10 +50,29 @@ func DeleteRevisions(r client.Client, ctx context.Context, partitionJob *webappv
 
 	deleteAllOptions := &client.DeleteAllOfOptions{ListOptions: client.ListOptions{Namespace: partitionJob.Namespace, LabelSelector: labelSelector}, DeleteOptions: client.DeleteOptions{}}
 
-	if err := r.DeleteAllOf(ctx, &apps.ControllerRevision{}, deleteAllOptions); err != nil {
+	revisionList := &apps.ControllerRevisionList{}
+
+	listOptions := &client.ListOptions{Namespace: partitionJob.Namespace, LabelSelector: labelSelector}
+
+	if err := r.List(ctx, revisionList, listOptions); err != nil {
 		return err
 	}
+	finalizerName := "webapp.my.domain.partitionjob/finalizer"
+	for index := range revisionList.Items {
+
+		controllerutil.RemoveFinalizer(&revisionList.Items[index], finalizerName)
+		if err := r.Update(ctx, &revisionList.Items[index]); err != nil {
+			return err
+		}
+
+		if err := r.Delete(ctx, &revisionList.Items[index], deleteAllOptions); err != nil {
```

Comment:
You can just remove the finalizer and don't need to call delete because after removing the finalizer the controller revisions will be deleted automatically if the deletiontimestamp is set.